components/alert/alert-toast.js

```diff
@@ -387,14 +387,19 @@ class AlertToast extends LitElement {
 		}
 	}
 
+	_onStateClosed() {
+		this._state = states.CLOSED;
+		this._totalSiblingHeightBelow = 0;
+		this._numAlertsBelow = 0;
+		this._closeClicked = false;
+		this.removeAttribute('role');
+	}
+
 	_onTransitionEnd() {
 		if (this._state === states.OPENING || this._state === states.SLIDING) {
 			this._state = states.OPEN;
 		} else if (this._state === states.CLOSING) {
-			this._state = states.CLOSED;
-			this._totalSiblingHeightBelow = 0;
-			this._closeClicked = false;
-			this._numAlertsBelow = 0;
+			this._onStateClosed();
```

Comment:
Another option for this could be updating `this._state` in these 2 places and then in `updated` updating the other properties when state becomes CLOSED. Let me know if there are preferences. In some quick testing both options seemed to cause the same number of renders.